### PR TITLE
Blogapp

### PR DESCRIPTION
**Pull request by acolombo1**
- Create two users with rails c.
- Create 5 posts written by the first user.
- Create 6 comments for the fourth post.
- Use other [CRUD methods] to find, update and delete entities.
- Create custom methods:
  - A method that returns the 3 most recent posts for a given user.
  - A method that updates the posts counter for a user.
  - A method that returns the 5 most recent comments for a given post.
  - A method that updates the comments counter for a post.
  - A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: 'author_id'
+  has_many :comments, foreign_key: 'author_id'
+  has_many :likes, foreign_key: 'author_id'
+  def last_3_posts
+    posts.limit(3).order(created_at: :desc)
+  end
```
Comment (brytebee) [inline app/models/user.rb]: - [x] **Optional:** Like the method you've written in Post class, the limit keyword is conventionally used at the end of your query.

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', counter_cache: 'posts_counter'
+  has_many :comments
+  has_many :likes
+  def last_5_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+end
```
Comment (brytebee) [inline app/models/post.rb:16]: - [x] **Required:** You are required to have a method that updates the counter when a new post is added. Kindly add the method.

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post, counter_cache: 'comments_counter'
+end
```
Comment (brytebee) [inline app/models/comment.rb:16]: - [x] **Required:** You are required to have a method that updates the counter when a new comment is added. Kindly add the method.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @acolombo1👋,

Good job so far coming so far with the progress you've made! 💪💪

## Highlights
You have done amazing in the following:
- Good commit messages and PR summary.✔️
- Associations are set up correctly.✔️
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AnselemOdims — APPROVED** [review]
### Hi  @acolombo1 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**